crates/holochain/src/conductor/entry_def_store.rs

```diff
@@ -129,12 +130,17 @@ impl BufferedStore for EntryDefBuf {
 pub(crate) async fn get_entry_def(
     entry_def_index: EntryDefIndex,
     zome: Zome,
-    dna_file: &DnaFile,
+    dna_def: &DnaDefHashed,
     conductor_api: &impl CellConductorApiT,
 ) -> EntryDefStoreResult<Option<EntryDef>> {
     // Try to get the entry def from the entry def store
     let key = EntryDefBufferKey::new(zome, entry_def_index);
-    let entry_def = { conductor_api.get_entry_def(&key).await };
+    let entry_def = conductor_api.get_entry_def(&key).await;
+    let dna_hash = dna_def.as_hash();
+    let dna_file = conductor_api
+        .get_dna(dna_hash)
+        .await
+        .ok_or_else(|| EntryDefStoreError::DnaFileMissing(dna_hash.clone()))?;
```

Comment:
This is the one and only place where we need to actually go from `DnaDef` to `DnaFile`